docs/specs/field_edit_markup.md

```diff
@@ -0,0 +1,1295 @@
+        <div class="ez-field-preview-tools">
+            <a href="/path/to/file">Download</a>
+            <label>
+                Remove <input type="checkbox" name="checkbox-auto-generated-name" value="1">
+            </label>
+        </div>
+    </div>
+    <div class="ez-field-edit-ui">
+        <input type="file" id="auto-generated-id" name="auto-generated-name">
+    </div>
+</div>
+```
+
+When set as required:
+
+As for others Field Type, the outermost `div` gets the `ez-field-edit-required`
+class and the file `input` gets the `required` attribute.
+
+## Float `ezfloat`
+
+### Options
+
+* Minimum value
+* Maximum value
+* Required
+
+### Markup
+
+Without any constraints:
+
+```html
+<div class="ez-field-edit ez-field-edit-ezfloat">
+    <div class="ez-field-edit-text-zone">
+        <label class="ez-field-definition-name" for="auto-generated-id">
+            Field Definition Name
+        </label>
+    </div>
+    <div class="ez-field-edit-ui">
+        <input type="number" step="any" id="auto-generated-id" name="auto-generated-name" value="">
+    </div>
+</div>
+```
+
+When set required and with a min and max value:
+
+```html
+<div class="ez-field-edit ez-field-edit-ezfloat ez-field-edit-required">
+    <div class="ez-field-edit-text-zone">
+        <label class="ez-field-definition-name" for="auto-generated-id">
+            Field Definition Name
+        </label>
+    </div>
+    <div class="ez-field-edit-ui">
+        <input type="number" step="any" min="5.2" max="10.6" required id="auto-generated-id" name="auto-generated-name" value="">
+    </div>
+</div>
+```
+
+## Map Location `ezgmaplocation`
+
+### Options
+
+* Required
+
+### Markup
+
+```html
+<div class="ez-field-edit ez-field-edit-ezgmaplocation">
+    <div class="ez-field-edit-text-zone">
+        <label class="ez-field-definition-name"><!-- no for attribute -->
+            Field Definition Name
+        </label>
+    </div>
+    <div class="ez-field-edit-ui">
+        <fieldset>
+            <div class="ez-sub-field ez-sub-field-latitude">
+                <div class="ez-sub-field-text-zone">
+                    <label class="ez-sub-field-name" for="auto-generated-id-lat">
+                        Latitude
+                    </label>
+                </div>
+                <div class="ez-sub-field-ui">
+                    <input type="number" step="any" min="-90" max="90" id="auto-generated-id-lat" name="auto-generated-name-lat">
+                </div>
+            </div>
+            <div class="ez-sub-field ez-sub-field-longitude">
+                <div class="ez-sub-field-text-zone">
+                    <label class="ez-sub-field-name" for="auto-generated-id-lon">
+                        Longitude
+                    </label>
+                </div>
+                <div class="ez-sub-field-ui">
+                    <input type="number" step="any" min="-90" max="90" id="auto-generated-id-lon" name="auto-generated-name-lon">
```

Comment:
with the server side generated markup only yes, then when we'll apply frontend enhancement with the ability the pick a Location in a map and while doing that we *can* restrict manual input (even if I don't really see the point in putting this restriction in place)